### DISCUSSION TITLE
Calling `Download a repository archive (zip)` API returns zip file without .git folder

### DISCUSSION THREAD

**Discussion by kernelguardian** [community/community#40512] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I was calling [this api](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#download-a-repository-archive-zip) which returns a zip folder of the repo. When I checked the folder contents it didn't have a .git folder. Is this the expected behaviour? If it is what should I do to fetch a repo with .git folder?

**Comment by airtower-luna** [comment, marked as answer]
That is expected behavior, the ZIP contains the state of the repository at a certain commit. You should clone the repository using Git instead. See: [Git Basics - Getting a Git Repository](https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository)

**Reply by kernelguardian** [reply]
What if I want to do that programmatically? Is it possible?

**Reply by airtower-luna** [reply]
Sure: Either use a Git library for the programming language you're using, or call the Git binary from your program. In the latter case be careful not to create a shell injection vulnerability.

**Reply by kernelguardian** [reply]
Thank you @airtower-luna

**Comment by Doctor2211** [comment]
I was calling this api which returns a zip folder of the repo. When I checked the folder contents it didn't have a .git folder[.](https://xn--220b953bd3h.com/) Is this the expected behaviour? If it is what should I do to fetch a repo with .git folder?
